### ISSUE TITLE
move/copy operation failed with permission denied

### ISSUE THREAD

**Issue by affeldt28** [HaveAGitGat/Tdarr#668]
**Describe the bug**
Can't copy back to original folder.

**To Reproduce**
- Install fresh tdarr
- setup tdarr user
- create service for systemd
- add smb mount in fstab
- create library 
- let its transcode

**Expected behavior**
Should copy/override original file with new transcoded file

**Please provide the following information:**
- Config files:
  - tdarr server (default config):
```json
{
  "serverPort": "8266",
  "webUIPort": "8265",
  "serverIP": "0.0.0.0",
  "handbrakePath": "",
  "ffmpegPath": "/usr/bin/ffmpeg",
  "logLevel": "INFO"
}
```
  -  fstab
```fstab
...
//jakku.alderaan.local/Plex  /mnt/Plex  cifs  _netdev,credentials=/root/.smb  0  0
```
  - /etc/systemd/system/tdarr-server.service
```service
[Unit]
Description=Tdarr Server
After=network.target

[Service]
Type=simple
ExecStart=/home/tdarr/Tdarr_Server/Tdarr_Server
Restart=always
RestartSec=10
TimeoutStartSec=30
RemainAfterExit=yes
User=tdarr
Group=tdarr

[Install]
WantedBy=default.target
```
  - tdarr node:
```json
{
  "nodeID": "coruscant",
  "serverIP": "192.168.2.105",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": "",
  "mkvpropeditPath": "",
  "pathTranslators": [
    {
      "server": "/mnt/Plex",
      "node": "x:\\"
    }
  ],
  "platform_arch": "win32_x64_docker_false",
  "logLevel": "INFO"
}
```

- Log files
  - serverlog:
```
[2022-08-03T22:08:14.154] [INFO] Tdarr_Server - ---------------Binary tests start----------------
[2022-08-03T22:08:14.157] [INFO] Tdarr_Server - handbrakePath:HandBrakeCLI
[2022-08-03T22:08:14.160] [INFO] Tdarr_Server - ffmpegPath:/usr/bin/ffmpeg
[2022-08-03T22:08:14.164] [INFO] Tdarr_Server - mkvpropedit:mkvpropedit
[2022-08-03T22:08:14.165] [INFO] Tdarr_Server - Binary test 1: handbrakePath working
[2022-08-03T22:08:14.165] [INFO] Tdarr_Server - Binary test 2: ffmpegPath working
[2022-08-03T22:08:14.165] [INFO] Tdarr_Server - ---------------Binary tests end-------------------
[2022-08-03T22:08:14.167] [INFO] Tdarr_Server - ---------------Scanner tests start----------------
[2022-08-03T22:08:14.168] [INFO] Tdarr_Server - Scanner test: Scanning file start
[2022-08-03T22:08:18.688] [INFO] Tdarr_Server - Scanner test: Scanning file end
[2022-08-03T22:08:18.688] [INFO] Tdarr_Server - Scanner test 1: FFprobe working
[2022-08-03T22:08:18.689] [INFO] Tdarr_Server - Scanner test 2: Exiftool working
[2022-08-03T22:08:18.689] [INFO] Tdarr_Server - Scanner test 3: Mediainfo working
[2022-08-03T22:08:18.689] [WARN] Tdarr_Server - Scanner test 4: CCExtractor not working:"/home/tdarr/Tdarr_Server/assets/app/ccextractor/ccextractor: error while loading shared libraries: libtesseract.so.4: cannot open shared object file: No such file or directory\n"
[2022-08-03T22:08:18.689] [INFO] Tdarr_Server - ---------------Scanner tests end----------------
[2022-08-03T22:08:18.691] [INFO] Tdarr_Server - Database has finished initialising
[2022-08-03T22:08:18.864] [INFO] Tdarr_Server - Updating plugins
[2022-08-03T22:08:20.277] [INFO] Tdarr_Server - Cloning plugins
[2022-08-03T22:08:20.526] [INFO] Tdarr_Server - Job report history size is within limit. Limit:10240 MiB, Size:42 MiB
[2022-08-03T22:08:20.746] [INFO] Tdarr_Server - Finished downloading plugins!
[2022-08-03T22:08:24.414] [INFO] Tdarr_Server - [4139]Plugin update finished
[2022-08-03T22:08:25.766] [INFO] Tdarr_Server - Node tdarr-server registered
[2022-08-03T22:10:55.255] [INFO] Tdarr_Server - Reading DB:__________(0%):1:SettingsGlobalJSONDB:/home/tdarr/server/Tdarr/DB2/SettingsGlobalJSONDB/afdf79ef24e7f6f173693057d0c2f3c07b0d3a3c.json:R/W Accessible
[2022-08-03T22:10:55.258] [INFO] Tdarr_Server - Reading DB:__________(0%):1:SettingsGlobalJSONDB:/home/tdarr/server/Tdarr/DB2/SettingsGlobalJSONDB/afdf79ef24e7f6f173693057d0c2f3c07b0d3a3c.json:Reading
[2022-08-03T22:10:55.259] [INFO] Tdarr_Server - Reading DB:__________(0%):1:SettingsGlobalJSONDB:/home/tdarr/server/Tdarr/DB2/SettingsGlobalJSONDB/afdf79ef24e7f6f173693057d0c2f3c07b0d3a3c.json:Adding
[2022-08-03T22:10:55.260] [INFO] Tdarr_Server - Reading DB:>>>>>>>>>>(100%):1:SettingsGlobalJSONDB::Reading done
[2022-08-03T22:10:55.260] [INFO] Tdarr_Server - Reading DB:__________(0%):1:LibrarySettingsJSONDB:/home/tdarr/server/Tdarr/DB2/LibrarySettingsJSONDB/efd816eb088a435e61a38bd02f07eede8ceca384.json:R/W Accessible
[2022-08-03T22:10:55.261] [INFO] Tdarr_Server - Reading DB:__________(0%):1:LibrarySettingsJSONDB:/home/tdarr/server/Tdarr/DB2/LibrarySettingsJSONDB/efd816eb088a435e61a38bd02f07eede8ceca384.json:Reading
[2022-08-03T22:10:55.261] [INFO] Tdarr_Server - Reading DB:__________(0%):1:LibrarySettingsJSONDB:/home/tdarr/server/Tdarr/DB2/LibrarySettingsJSONDB/efd816eb088a435e61a38bd02f07eede8ceca384.json:Adding
[2022-08-03T22:10:55.261] [INFO] Tdarr_Server - Reading DB:>>>>>>>>>>(100%):1:LibrarySettingsJSONDB::Reading done
[2022-08-03T22:10:55.262] [INFO] Tdarr_Server - Reading DB:__________(0%):1:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/1b7aa1e081ce8de0de0bdc4db00e77536d4d4b28.json:R/W Accessible
[2022-08-03T22:10:55.262] [INFO] Tdarr_Server - Reading DB:__________(0%):1:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/1b7aa1e081ce8de0de0bdc4db00e77536d4d4b28.json:Reading
[2022-08-03T22:10:55.262] [INFO] Tdarr_Server - Reading DB:__________(0%):1:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/1b7aa1e081ce8de0de0bdc4db00e77536d4d4b28.json:Adding
[2022-08-03T22:10:55.263] [INFO] Tdarr_Server - Reading DB:>>>>>_____(50%):2:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/523087abefe7f2ab46bc5b09935a4cb8b3391043.json:R/W Accessible
[2022-08-03T22:10:55.263] [INFO] Tdarr_Server - Reading DB:>>>>>_____(50%):2:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/523087abefe7f2ab46bc5b09935a4cb8b3391043.json:Reading
[2022-08-03T22:10:55.263] [INFO] Tdarr_Server - Reading DB:>>>>>_____(50%):2:NodeJSONDB:/home/tdarr/server/Tdarr/DB2/NodeJSONDB/523087abefe7f2ab46bc5b09935a4cb8b3391043.json:Adding
[2022-08-03T22:10:55.263] [INFO] Tdarr_Server - Reading DB:>>>>>>>>>>(100%):2:NodeJSONDB::Reading done
[2022-08-03T22:10:55.264] [INFO] Tdarr_Server - Reading DB:__________(0%):1:StatisticsJSONDB:/home/tdarr/server/Tdarr/DB2/StatisticsJSONDB/3d18b2ea89fa5357cf6423b4e6985b46fa10195e.json:R/W Accessible
[2022-08-03T22:10:55.264] [INFO] Tdarr_Server - Reading DB:__________(0%):1:StatisticsJSONDB:/home/tdarr/server/Tdarr/DB2/StatisticsJSONDB/3d18b2ea89fa5357cf6423b4e6985b46fa10195e.json:Reading
[2022-08-03T22:10:55.264] [INFO] Tdarr_Server - Reading DB:__________(0%):1:StatisticsJSONDB:/home/tdarr/server/Tdarr/DB2/StatisticsJSONDB/3d18b2ea89fa5357cf6423b4e6985b46fa10195e.json:Adding
[2022-08-03T22:10:55.265] [INFO] Tdarr_Server - Reading DB:>>>>>>>>>>(100%):1:StatisticsJSONDB::Reading done
[2022-08-03T22:10:55.530] [INFO] Tdarr_Server - /home/tdarr/configs/Tdarr_Server_Config.json
[2022-08-03T22:10:55.545] [INFO] Tdarr_Server - {
  serverPort: '8266',
  webUIPort: '8265',
  serverIP: '0.0.0.0',
  handbrakePath: '',
  ffmpegPath: '/usr/bin/ffmpeg',
  logLevel: 'INFO'
}
[2022-08-03T22:10:56.196] [INFO] Tdarr_Server - version: 2.00.18
[2022-08-03T22:10:56.197] [INFO] Tdarr_Server - linux_x64_docker_false
[2022-08-03T22:10:56.197] [INFO] Tdarr_Server - Preparing environment
[2022-08-03T22:10:56.198] [INFO] Tdarr_Server - Tdarr started.
[2022-08-03T22:10:56.198] [INFO] Tdarr_Server - Tdarr documents folder:/home/tdarr/server
[2022-08-03T22:10:56.198] [INFO] Tdarr_Server - Checking directories
[2022-08-03T22:10:56.201] [INFO] Tdarr_Server - Tdarr_Server listening at http://0.0.0.0:8266
[2022-08-03T22:10:56.201] [TRACE] Tdarr_Server -
    --------------------------------------------------------------------------
    --------------------------------------------------------------------------

    Tdarr_WebUI running at http://localhost:8265

    --------------------------------------------------------------------------
    --------------------------------------------------------------------------

[2022-08-03T22:10:56.310] [INFO] Tdarr_Server - Node coruscant registered
[2022-08-03T22:10:56.399] [INFO] Tdarr_Server - Node tdarr-server registered
[2022-08-03T22:10:57.862] [INFO] Tdarr_Server - ---------------Binary tests start----------------
[2022-08-03T22:10:57.864] [INFO] Tdarr_Server - handbrakePath:HandBrakeCLI
[2022-08-03T22:10:57.865] [INFO] Tdarr_Server - ffmpegPath:/usr/bin/ffmpeg
[2022-08-03T22:10:57.865] [INFO] Tdarr_Server - mkvpropedit:mkvpropedit
[2022-08-03T22:10:57.865] [INFO] Tdarr_Server - Binary test 1: handbrakePath working
[2022-08-03T22:10:57.866] [INFO] Tdarr_Server - Binary test 2: ffmpegPath working
[2022-08-03T22:10:57.866] [INFO] Tdarr_Server - ---------------Binary tests end-------------------
[2022-08-03T22:10:57.867] [INFO] Tdarr_Server - ---------------Scanner tests start----------------
[2022-08-03T22:10:57.868] [INFO] Tdarr_Server - Scanner test: Scanning file start
[2022-08-03T22:11:02.139] [INFO] Tdarr_Server - Scanner test: Scanning file end
[2022-08-03T22:11:02.139] [INFO] Tdarr_Server - Scanner test 1: FFprobe working
[2022-08-03T22:11:02.139] [INFO] Tdarr_Server - Scanner test 2: Exiftool working
[2022-08-03T22:11:02.140] [INFO] Tdarr_Server - Scanner test 3: Mediainfo working
[2022-08-03T22:11:02.140] [WARN] Tdarr_Server - Scanner test 4: CCExtractor not working:"/home/tdarr/Tdarr_Server/assets/app/ccextractor/ccextractor: error while loading shared libraries: libtesseract.so.4: cannot open shared object file: No such file or directory\n"
[2022-08-03T22:11:02.140] [INFO] Tdarr_Server - ---------------Scanner tests end----------------
[2022-08-03T22:11:02.142] [INFO] Tdarr_Server - Database has finished initialising
[2022-08-03T22:11:02.315] [INFO] Tdarr_Server - Updating plugins
[2022-08-03T22:11:03.183] [INFO] Tdarr_Server - Cloning plugins
[2022-08-03T22:11:03.563] [INFO] Tdarr_Server - Finished downloading plugins!
[2022-08-03T22:11:03.906] [INFO] Tdarr_Server - Job report history size is within limit. Limit:10240 MiB, Size:42 MiB
[2022-08-03T22:11:07.155] [INFO] Tdarr_Server - [3974]Plugin update finished
```
- worker log:
```
[2022-08-03T21:52:20.433] [INFO] Tdarr_Node - C:/Users/user/tdarr/configs/Tdarr_Node_Config.json
[2022-08-03T21:52:20.446] [INFO] Tdarr_Node - {
  nodeID: 'coruscant',
  serverIP: '192.168.2.105',
  serverPort: '8266',
  handbrakePath: '',
  ffmpegPath: '',
  mkvpropeditPath: '',
  pathTranslators: [ { server: '/mnt/Plex', node: 'x:\\' } ],
  platform_arch: 'win32_x64_docker_false',
  logLevel: 'INFO'
}
[2022-08-03T21:52:20.612] [INFO] Tdarr_Node - version: 2.00.18
[2022-08-03T21:52:20.612] [INFO] Tdarr_Node - win32_x64_docker_false
[2022-08-03T21:52:20.613] [INFO] Tdarr_Node - Starting Tdarr_Node
[2022-08-03T21:52:20.613] [INFO] Tdarr_Node - Preparing environment
[2022-08-03T21:52:20.614] [INFO] Tdarr_Node - Path translator: Checking Node path: x:\
[2022-08-03T21:52:20.615] [INFO] Tdarr_Node - Updating plugins
[2022-08-03T21:52:20.761] [INFO] Tdarr_Node - Path translator: Success: Node path can be accessed: x:\
[2022-08-03T21:52:20.787] [INFO] Tdarr_Node - Node connected & registered
[2022-08-03T21:52:20.982] [INFO] Tdarr_Node - Cloning plugins
[2022-08-03T21:52:21.681] [INFO] Tdarr_Node - Finished downloading plugins!
[2022-08-03T21:52:22.042] [INFO] Tdarr_Node - ---------------Binary tests start----------------
[2022-08-03T21:52:22.042] [INFO] Tdarr_Node - handbrakePath:C:/Users/user/tdarr/Tdarr_Node/assets/app/HandBrakeCLI.exe
[2022-08-03T21:52:22.043] [INFO] Tdarr_Node - ffmpegPath:C:/Users/user/tdarr/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe
[2022-08-03T21:52:22.043] [INFO] Tdarr_Node - mkvpropedit:C:/Users/user/tdarr/Tdarr_Node/assets/app/mkvpropedit/windows/mkvpropedit.exe
[2022-08-03T21:52:22.043] [INFO] Tdarr_Node - Binary test 1: handbrakePath working
[2022-08-03T21:52:22.043] [INFO] Tdarr_Node - Binary test 2: ffmpegPath working
[2022-08-03T21:52:22.045] [INFO] Tdarr_Node - ---------------Binary tests end-------------------
[2022-08-03T21:52:26.545] [INFO] Tdarr_Node - [5565]Plugin update finished
[2022-08-03T22:08:02.674] [INFO] Tdarr_Node - Node connected & registered
```
  
- Worker error
  
```
2022-08-03T22:11:20.582Z 4sRtS1E7z3A:[Step S02] Beginning move/copy operation2022-08-03T22:11:20.582Z 4sRtS1E7z3A:Calculating old and new sizes of the following files
2022-08-03T22:11:20.583Z 4sRtS1E7z3A:/mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13.mkv
2022-08-03T22:11:20.583Z 4sRtS1E7z3A:/mnt/Plex/cache/XXX - S05E13-TdarrCacheFile-uQ6IFAGY76.mkv
2022-08-03T22:11:20.584Z 4sRtS1E7z3A:Old size 0.5538589069619775. New size 0.5835016891360283
2022-08-03T22:11:20.584Z 4sRtS1E7z3A:Folder to folder conversion is off
2022-08-03T22:11:20.585Z 4sRtS1E7z3A:New file path /mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv
2022-08-03T22:11:20.586Z 4sRtS1E7z3A:Ensuring output folder path exists /mnt/Plex/TV Shows/Other/XXX/Staffel 5
2022-08-03T22:11:21.586Z 4sRtS1E7z3A:Calculating cache file size in bytes
2022-08-03T22:11:21.587Z 4sRtS1E7z3A:626530168
2022-08-03T22:11:21.587Z 4sRtS1E7z3A:Attempting to move cache file to output folder, method 1
2022-08-03T22:11:21.588Z 4sRtS1E7z3A:File move error: Error: EACCES: permission denied, rename '/mnt/Plex/cache/XXX - S05E13-TdarrCacheFile-uQ6IFAGY76.mkv' -> '/mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv'
2022-08-03T22:11:21.588Z 4sRtS1E7z3A:File move error: Cache file /mnt/Plex/cache/XXX - S05E13-TdarrCacheFile-uQ6IFAGY76.mkv (626530168 bytes) does not match size of new cache file /mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv (0 bytes)
2022-08-03T22:11:21.589Z 4sRtS1E7z3A:Attempting to move cache file to output folder, method 2
2022-08-03T22:11:21.589Z 4sRtS1E7z3A:File move error: Cache file /mnt/Plex/cache/XXX - S05E13-TdarrCacheFile-uQ6IFAGY76.mkv (626530168 bytes) does not match size of new cache file /mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv (0 bytes)
2022-08-03T22:11:21.590Z 4sRtS1E7z3A:Attempting to copy
2022-08-03T22:11:22.591Z 4sRtS1E7z3A:File copy error: Error: EACCES: permission denied, open '/mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv'
2022-08-03T22:11:23.593Z 4sRtS1E7z3A:Moving/Copying item [-error-]: Cache file /mnt/Plex/cache/XXX - S05E13-TdarrCacheFile-uQ6IFAGY76.mkv (626530168 bytes) does not match size of new cache file /mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv (0 bytes)
2022-08-03T22:11:23.593Z 4sRtS1E7z3A:Performing clean up on file: /mnt/Plex/TV Shows/Other/XXX/Staffel 5/XXX - S05E13-TdarrCacheFile-v_lOprrZj_.mkv
2022-08-03T22:11:23.594Z 4sRtS1E7z3A:Can retry copying in staging section on Tdarr tab
```

 - Server-OS: openSuse 15.3
 - Tdarr-Version 2.00.18
 - Node-OS: Win10

**Additional context**
The user for the smb share has full write & read access. I also tried to run the service with root, but no difference.
Hope anyone can help me.

**Comment by HaveAGitGat** [comment]
Unfortunately permissions issue are a PIA. Just to check, you're not running the Node as admin right? As that can prevent writing to the share.

**Comment by affeldt28** [comment]
Yes, I created a user called 'tdarr'. I tried running the service as 'root', but unfortunately I got the same results. Then I tried running the move command by hand as root, but this didn't work either. So there must be a problem with the account used for the share.
The problem was that I created the files on the share with a different user. Even though the tdarr user has full write and read permission, he wasn't able to override my files.

Thanks for your time. Even though this wasn't a Tdarr issue and just my stupidity.
